spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/OpenLineageIntegrationTestBase.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery.integration;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.spark.bigquery.integration.SparkBigQueryIntegrationTestBase.TestDataset;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.SparkSession;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+
+public class OpenLineageIntegrationTestBase {
+  @ClassRule
+  public static TestDataset testDataset = new SparkBigQueryIntegrationTestBase.TestDataset();
+
+  @ClassRule public static CustomSessionFactory sessionFactory = new CustomSessionFactory();
+
+  protected SparkSession spark;
+  protected String testTable;
+  protected File lineageFile;
+
+  public OpenLineageIntegrationTestBase() {
+    this.spark = sessionFactory.spark;
+    this.lineageFile = sessionFactory.lineageFile;
+  }
+
+  @Before
+  public void createTestTable() {
+    testTable = "test_" + System.nanoTime();
+  }
+
+  protected static class CustomSessionFactory extends ExternalResource {
+    SparkSession spark;
+    File lineageFile;
+
+    @Override
+    protected void before() throws Throwable {
+      lineageFile = File.createTempFile("openlineage_test_" + System.nanoTime(), ".log");
+      lineageFile.deleteOnExit();
+      spark =
+          SparkSession.builder()
+              .master("local")
+              .appName("openlineage_test_bigquery_connector")
+              .config("spark.ui.enabled", "false")
+              .config("spark.default.parallelism", 20)
+              .config("spark.extraListeners", "io.openlineage.spark.agent.OpenLineageSparkListener")
+              .config("spark.openlineage.transport.type", "file")
+              .config("spark.openlineage.transport.location", lineageFile.getAbsolutePath())
+              .getOrCreate();
+      spark.sparkContext().setLogLevel("WARN");
+    }
+  }
+
+  private List<JSONObject> parseEventLogs(File file) throws Exception {
+    Scanner scanner = new Scanner(file);
```

Comment:
please use try-with-resources block to auto-close this scanner.

Reply:
updated, thanks